src/PDOAdapter.php

```diff
@@ -86,45 +86,38 @@ public function inTransaction()
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId($name = null)
+    public function lastInsertId(?string $name = null)
     {
         return $this->pdo->lastInsertId();
     }
 
     /**
      * {@inheritdoc}
      */
-    public function prepare($statement)
+    public function prepare(string $statement, array $options = [])
     {
         /** @psalm-var PDOStatement|false */
-        return $this->pdo->prepare($statement);
+        return $this->pdo->prepare($statement, $options);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function query($statement, $param1 = null, $param2 = null, $param3 = null)
+    public function query(string $query, ?int $fetchMode = null, ...$fetchModeArgs)
     {
-        if ($param1 === null || $param2 === null) {
-            /** @psalm-var PDOStatement|false */
-            return $this->pdo->query($statement);
-        }
-
-        if ($param3 === null) {
-            /** @psalm-var PDOStatement|false */
-            return $this->pdo->query($statement, $param1, $param2);
-        }
-
-        /** @psalm-var PDOStatement|false */
-        return $this->pdo->query($statement, $param1, $param2, $param3);
+        /**
+         * @psalm-var PDOStatement|false
+         * @psalm-suppress PossiblyNullArgument
```

Comment:
This psalm-suppress is needed until the next version of psalm is released.
See https://github.com/vimeo/psalm/pull/5100